### PR TITLE
DEV: Convert edit-topic-timer-form to glimmer

### PR DESCRIPTION
**Pull request by CvX**

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/edit-topic-timer-form.js [hunk 1]
```diff
@@ -17,73 +18,101 @@ import {
...
-    KeyboardShortcuts.pause();
-    this.set("_itsatrap", new ItsATrap());
-
-    this.set("duration", this.initialDuration);
-  },
+  @tracked timerType;
 
-  get initialDuration() {
-    if (!this.useDuration || !this.topicTimer.duration_minutes) {
+  duration = (() => {
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/edit-topic-timer-form.js:29]: dunno if this pattern is good 😉 

I think I like it better than the previous

```js
duration = this.initialDuration;

get initialDuration() {
  // a getter that's used only above
}
```

but maybe it'd be cleaner to just do this in the constructor 🤷

Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/components/edit-topic-timer-form.js:29]: Yeah agreed - the current pattern is cool, but also very surprising!

But also... having a getter feels weird. So maybe a function instead? `calculateInitialDuration()`?

```js
duration = this.#calculateInitialDuration();

#calculateInitialDuration(){
  //...
}
```

?

**Review by davidtaylorhq — APPROVED** [review]
LGTM, pending that field initializer change 🚀

File: app/assets/javascripts/discourse/app/components/edit-topic-timer-form.js (hunk 1 above)
Comment (CvX) [reply app/assets/javascripts/discourse/app/components/edit-topic-timer-form.js]: lol, the prop was unused 😄 the best code is deleted code 😛

**Pull request merged**